tests/Common/ConfigurationMetadataTest.php

```diff
@@ -0,0 +1,405 @@
+                ]
+            ]);
+        $branchComponents->addConfigurationMetadata($configurationMetadataOptions);
+
+        $branchComponents->resetToDefault('transformation', 'main-1');
+
+        $listConfigurationMetadata = $branchComponents->listConfigurationMetadata((new ListConfigurationMetadataOptions())
+            ->setComponentId('transformation')
+            ->setConfigurationId('main-1'));
+        self::assertCount(3, $listConfigurationMetadata);
+
+        $this->validateMetadataEquality(self::TEST_METADATA[0], $listConfigurationMetadata[0]);
+        $this->validateMetadataEquality($updatedMetadata[0], $listConfigurationMetadata[1]);
+        $this->validateMetadataEquality($moreMetadata[0], $listConfigurationMetadata[2]);
+    }
+
+    public function testConfigMetadataRestrictionsForReadOnlyUser()
+    {
+        $guestClient = $this->getClientForToken(STORAGE_API_READ_ONLY_TOKEN);
+
+        // create new configurations in main branch
+        $components = new Components($this->_client);
+        $transformationMain1Options = (new Configuration())
+            ->setComponentId('transformation')
+            ->setConfigurationId('main-1')
+            ->setName('Main 1')
+            ->setConfiguration(['test' => 'false']);
+
+        $components->addConfiguration($transformationMain1Options);
+        $configurationMetadataOptions = (new ConfigurationMetadata($transformationMain1Options))
+            ->setMetadata(self::TEST_METADATA);
+
+        try {
+            $components = new Components($guestClient);
+            $components->addConfigurationMetadata($configurationMetadataOptions);
+            $this->fail('should fail, insufficiently permission');
+        } catch (ClientException $e) {
+            $this->assertContains('Configuration manipulation is restricted for your user role ', $e->getMessage());
+            $this->assertSame(403, $e->getCode());
+        }
+    }
+
+    private function validateMetadataEquality(array $expected, array $actual)
+    {
+        foreach ($expected as $key => $value) {
+            self::assertArrayHasKey($key, $actual);
+            self::assertSame($value, $actual[$key]);
+        }
+        self::assertArrayHasKey('timestamp', $actual);
+    }
+
+    private function createConfiguration($components, $componentId, $configurationId, $name = 'Main-1')
+    {
+        $configurationOptions = (new Configuration())
+            ->setComponentId($componentId)
+            ->setConfigurationId($configurationId)
+            ->setName($name);
+
+        $components->addConfiguration($configurationOptions);
+        return $configurationOptions;
+    }
+
+    private function prepareBranchesForTests(DevBranches $devBranch, $branchName, $createDevBranch = true)
+    {
+        //clean devbranch
+        $this->deleteBranchesByPrefix($devBranch, $branchName);
+
+        $branchesList = $devBranch->listBranches();
+
+        //get default branchId
+        $defaultBranch = null;
```

Comment:
čistě pro pořádek - mělo by to být defaultBranchId, ať je jasné, že je to jen IDčko, ale to je drobnost